### DISCUSSION TITLE
error using Zig compiler: backend compiler error

### DISCUSSION THREAD

**Discussion by rakshasha-medhi** [vlang/v#20837] [Installation Issues]
Hi everyone!

I'm trying install V but I also have Zig installed in my windows system, which I think I have set to use Zig compiler as the default C/C++ compiler. When I tried to run: 
./make.bat or 
v.exe symlink or 
gcc -std=gnu11 -w -o v.exe v.c

I always encountered this error:
```
cannot compile <path to vlang directory>\Vlang\v\cmd\tools\vsymlink.v`: 1
In file included from <path to Zig compiler>/zig-windows-x86_64-0.11.0/lib/include/inttypes.h:21:
In file included from <path to vlang directory>/vlang/v/thirdparty/tcc/include/inttypes.h:11:
In file included from <path to vlang directory>/vlang/v/thirdparty/tcc/include/_mingw.h:27:
<path to Zig compiler>/zig-windows-x86_64-0.11.0/lib/include/stddef.h:17: error: function pointer expected
builder error:
==================
C error. This should never happen.
This is a compiler bug, please report it using `v bug file.v`.
```

I uses Zig daily so the toolchain is important to me...uninstalling it is not an option.
I was contemplating to install a new mingw for C/C++ compiler but hesitated due to the potential conflict of multiple C/C++ compilers in the same system.

What are the alternatives to make this work? I want to try V especially the V to C converter tools.
Should I flag this issue in Zig's github as well? 
Please advise, cheers!

**Comment by JalonSolov** [comment]
Take a look at https://github.com/malisipi/vzcc - this is the closest anyone has come to using zig with V.

**Reply by rakshasha-medhi** [reply]
thanks for the reference.
Unfortunately, in the first step the requirement says I need to have V and Zig installed, and then install the vzcc.
--> the issue is I can't even install V coz the Zig compiler issue, so I can't even pass step #1.

I might look around over what are the side effects to having multiple C/C++ compilers in a system. If the side effects are not that great, then I might go through the multiple compilers route. 
Cheers!

**Reply by JalonSolov** [reply]
Well, V comes with tcc (tiny c compiler), which is what it uses by default.  That will work to compile V.

**Reply by JalonSolov** [reply]
Just follow these instructions to install V: https://github.com/vlang/v?tab=readme-ov-file#installing-v-from-source

**Reply by rakshasha-medhi** [reply]
Yeah TCC did not work as well coz it failed when I run .\make.bat --> the same error as previously.

But I figured it out:
I installed latest GCC and everything went smooth. I managed to install V then compile and run V files without issue.
Thanks! looking forward to have a lot of fun with V!
